src/store/middleware/model-poller.ts

```diff
@@ -285,17 +286,30 @@ export const modelPollerMiddleware: Middleware<
     ) {
       const { payload } = action;
       const conn = controllers.get(payload.wsControllerURL);
-      const response = await setModelSharingPermissions(
-        payload.wsControllerURL,
-        payload.modelUUID,
-        conn,
-        payload.user,
-        payload.permissionTo,
-        payload.permissionFrom,
-        payload.action,
-        reduxStore.dispatch,
-      );
-      return response;
+      try {
+        const response = await setModelSharingPermissions(
+          payload.wsControllerURL,
+          payload.modelUUID,
+          conn,
+          payload.user,
+          payload.permissionTo,
+          payload.permissionFrom,
+          payload.action,
+          reduxStore.dispatch,
+        );
+        return response;
+      } catch (error) {
+        console.error(
+          `Could not grant sharing permissions to model ${payload.modelUUID}`,
+          error,
+        );
+        reduxStore.dispatch(
+          jujuActions.updateModelsError({
```

Comment:
Instead of handling the error inside the middleware and storing it in the model errors I think we should allow this to throw which will return the error to `ShareModel` which has as try catch around the `updatePermissions` dispatch and displays the error in a toast. I think you can remove your changes and it will handle the error correctly.